mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/MantisAkkaRpcSystemLoader.java

```diff
@@ -16,24 +16,71 @@
 
 package io.mantisrx.server.core;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ServiceLoader;
+import java.util.UUID;
 import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.CoreOptions;
+import org.apache.flink.core.classloading.ComponentClassLoader;
 import org.apache.flink.runtime.rpc.RpcSystem;
 import org.apache.flink.runtime.rpc.RpcSystemLoader;
-import org.apache.flink.runtime.rpc.akka.AkkaRpcSystem;
+import org.apache.flink.util.IOUtils;
 
 /**
  * RpcSystemLoader for mantis task executor and other services that need to expose an RPC API.
  * This particular implementation uses the akka RPC implementation under the hood.
  */
 public class MantisAkkaRpcSystemLoader implements RpcSystemLoader {
 
-    private static final RpcSystem INSTANCE = new AkkaRpcSystem();
+    private static final RpcSystem INSTANCE = createRpcSystem();
+
+    public static RpcSystem getInstance() {
+        return INSTANCE;
+    }
+
     @Override
     public RpcSystem loadRpcSystem(Configuration config) {
         return INSTANCE;
     }
 
-    public static RpcSystem getInstance() {
-        return INSTANCE;
+    private static RpcSystem createRpcSystem() {
+        try {
+            final ClassLoader flinkClassLoader = RpcSystem.class.getClassLoader();
+
+            final Path tmpDirectory = Paths.get(System.getProperty("java.io.tmpdir"));
+            Files.createDirectories(tmpDirectory);
+            final Path tempFile =
+                Files.createFile(
+                    tmpDirectory.resolve("flink-rpc-akka_" + UUID.randomUUID() + ".jar"));
+
+            final InputStream resourceStream =
+                flinkClassLoader.getResourceAsStream("flink-rpc-akka.jar");
+            if (resourceStream == null) {
+                throw new RuntimeException(
+                    "Akka RPC system could not be found. If this happened while running a test in the IDE,"
+                        + "run the process-resources phase on flink-rpc/flink-rpc-akka-loader via maven.");
+            }
+
+            IOUtils.copyBytes(resourceStream, Files.newOutputStream(tempFile));
+
+            final ComponentClassLoader componentClassLoader = new ComponentClassLoader(
+                new URL[] {tempFile.toUri().toURL()},
+                flinkClassLoader,
+                CoreOptions.parseParentFirstLoaderPatterns(
+                    "org.slf4j;org.apache.log4j;org.apache.logging;org.apache.commons.logging;ch.qos.logback;io.mantisrx.server.worker;io.mantisrx.server.core;io.mantisrx", ""),
```

Comment:
Can you add some comments about how/where this list is composed? BTW is 'logback' needed as we don't use it?

Reply:
+1

Reply:
+1